bundles/binding/org.openhab.binding.sonance/src/main/java/org/openhab/binding/sonance/internal/SonanceBinding.java

```diff
@@ -0,0 +1,447 @@
+			return;
+		}
+
+		logger.info("Refreshing all items");
+
+		List<String> offlineEndPoints = new ArrayList<String>();
+
+		for (SonanceBindingProvider provider : providers) {
+			for (String itemName : provider.getItemNames()) {
+				String group = provider.getGroup(itemName);
+				String ip = provider.getIP(itemName);
+				int port = provider.getPort(itemName);
+
+				String key = ip + ":" + port;
+				if (!offlineEndPoints.contains(key)) {
+					try {
+						if (!socketCache.containsKey(key)) {
+							socketCache.put(key, new Socket(ip, port));
+							outputStreamCache.put(key, new DataOutputStream(
+									socketCache.get(key).getOutputStream()));
+							bufferedReaderCache.put(key, new BufferedReader(
+									new InputStreamReader(socketCache.get(key)
+											.getInputStream())));
+							logger.debug("New socket created");
+						}
+
+						if (provider.isMute(itemName))
+							sendMuteCommand(itemName, "FF550212" + group,
+									outputStreamCache.get(key),
+									bufferedReaderCache.get(key));
+						else if (provider.isVolume(itemName))
+							sendVolumeCommand(itemName, "FF550210" + group,
+									outputStreamCache.get(key),
+									bufferedReaderCache.get(key));
+					} catch (UnknownHostException e) {
+						logger.error("UnknownHostException occured.");
+					} catch (IOException e) {
+						logger.debug("Amplifier is offline, status can't be updated at this moment.");
+						try {
+							socketCache.get(key).close();
+						} catch (Exception ex) {
+						}
+						socketCache.remove(key);
+						outputStreamCache.remove(key);
+						bufferedReaderCache.remove(key);
+						offlineEndPoints.add(key); // Stop trying to fetch other
+													// values from this end
+													// point until next execute
+													// cycle
+					}
+				}
+			}
+		}
+	}
+
+	@Override
+	public void bindingChanged(BindingProvider provider, String itemName) {
+		super.bindingChanged(provider, itemName);
+
+		for (Map.Entry<String, Socket> entry : socketCache.entrySet())
+			try {
+				entry.getValue().close();
+			} catch (IOException e) {
+				logger.error("Can't close a socket when binding changed.");
+			}
+
+		// Cleanup all sockets
+		socketCache.clear();
+		outputStreamCache.clear();
+		bufferedReaderCache.clear();
+	}
+
+	@Override
+	protected void internalReceiveCommand(String itemName, Command command) {
+		logger.info("Command received ({}, {})", itemName, command);
+
+		SonanceBindingProvider provider = findFirstMatchingBindingProvider(itemName);
+		String group = provider.getGroup(itemName);
+		String ip = provider.getIP(itemName);
+		int port = provider.getPort(itemName);
+
+		Socket s = null;
+		try {
+			s = new Socket(ip, port);
+			DataOutputStream outToServer = new DataOutputStream(
+					s.getOutputStream());
+			BufferedReader i = new BufferedReader(new InputStreamReader(
+					s.getInputStream()));
+
+			if (provider.isMute(itemName))
+				switch (command.toString()) {
```

Comment:
in general all OH1 bindings should be compatible to java1.5 so please change into if-elseif-else condition